### ISSUE TITLE
push_notebook updates at most one plot

### ISSUE THREAD

**Issue by rs2** [bokeh/bokeh#6718]
### Environment
- bokeh=0.12.6=py36_0
- jupyter_client=5.1.0=py36_0
- notebook=5.0.0=py36_0
- python=3.6.2=0

### Observed
```python
push_notebook(handle=h1)
push_notebook(handle=h2)
```
updates only the plot referenced by `h1`, but not `h2`.

### Expected
Both plots get updated.

### Root cause
1. `h1.doc` and `h2.doc` are the same document.
1. `Document._compute_patch_between_json()` leads to `_json` being stored as a class not instance attribute.
  2.1. Should `bokeh.io._CommsHandle._json` be a class rather than an instance attribute?

#### Setup
1. `conda create -n bokeh6718 jupyter bokeh -y & source activate bokeh6718`
#### Notebook snippet
```python
from bokeh.io import push_notebook, show, output_notebook
from bokeh.plotting import figure, ColumnDataSource

output_notebook(hide_banner=True)

source = ColumnDataSource(dict(foo=[1, 2], bar=[42, 43]))

p1 = figure()
p1.circle('foo', 'bar', source=source)

p2 = figure()
p2.circle('foo', 'bar', source=source)

h1 = show_notebook(p1, notebook_handle=True)
h2 = show_notebook(p2, notebook_handle=True)

source.data['foo'][1] = 13
push_notebook(handle=h2)
push_notebook(handle=h1)

```

**Comment by bryevdv** [comment]
Possibly related to #5415

The current implementation of `push_notebook` is probably unsalvageable. I have a plan in mind to improve `push_notebook` but there are several steps that have to happen first:

* Bokeh updated to work with JupyterLab, and dust settles from that
* Have embedded apps in notebooks use notebook comms instead of a custom web socket
* implement batching, non-immediate update mode
* Re-implement `push_notebook` to use a bokeh server and standard protocol behind the scenes (batch changes until `push_notebook` is called)

I think this would put `push_notebook` on very solid footing but there's quite a bit of groundwork to lay to get there.

**Comment by bryevdv** [comment]
This is fixed by #6945
